OpenRA.Mods.Common/Installer/SourceActions/CopySourceAction.cs

```diff
@@ -0,0 +1,55 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using OpenRA.Mods.Common.Widgets.Logic;
+
+namespace OpenRA.Mods.Common.Installer
+{
+	public class CopySourceAction : ISourceAction
+	{
+		public void Run(MiniYaml actionYaml, string path, ModData modData, List<string> extracted, Action<string> updateMessage)
+		{
+			var sourceDir = Path.Combine(path, actionYaml.Value);
+			foreach (var node in actionYaml.Nodes)
+			{
+				var sourcePath = Path.Combine(sourceDir, node.Value.Value).Replace("/./", "/");
```

Comment:
Why is the `.Replace` needed? This seems like a bodge?

Also, what about paths relative to a named OpenRA directory?

Reply:
I had a problem with getting a path which has /./ somewhere in the middle, as the copy yaml node has a value of "." which is somewhat put in the middle. Ill re-check wether this problem still persist and give some additional insights on that...